cocos/core/scene-graph/base-node.ts

```diff
@@ -336,7 +328,19 @@ export class BaseNode extends CCObject implements ISchedulable {
 
     protected _siblingIndex = 0;
 
-    protected _registerIfAttached = !EDITOR ? undefined : function (this: BaseNode, register) {
+    /**
+     * Set `_scene` field of this node.
+     * The derived `Scene` overrides this method to behavior differently.
+     */
+    protected _updateScene () {
+        if (this._parent == null) {
```

Comment:
As doc says, `Scene._udpateScene` override this implementation. Then we eliminate a potential circular reference.

Reply:
Or we can just set `scene._scene = scene;`